ldap_auth_provider.py

```diff
@@ -123,7 +123,10 @@ async def check_auth(
         # Default display name for the user, if a new account is registered.
         default_display_name = username
         # Local part of Matrix ID which will be used in registration process
-        localpart = username
+        if username.isnumeric():
+            localpart = f"U{username}"
+        else:
+            localpart = username
```

Comment:
I think this implies conflicts if you have both `1234` and `U1234` as unique usernames in your system?

Reply:
Yes, that is correct. I cannot imagine how to solve this problem at this time.

Reply:
This can do the trick. Also, I will add one more change to overcome a bug while getting encryption keys. When you try to download backup keys, it asks for the password. Even if you enter it correctly, synapse says it is invallid. This is because the module tries to send the username with the prefix, however there is no match in LDAP for the username with prefix. I had also fixed this problem and will publish another change.